src/Nethermind/Nethermind.Evm/TransactionProcessor.cs

```diff
@@ -207,13 +207,9 @@ private void Execute(Transaction transaction, BlockHeader block, ITxTracer txTra
             Address? recipientOrNull = null;
             try
             {
-                Address recipient;
+                Address recipient = transaction.GetRecipient(transaction.IsContractCreation ? _stateProvider.GetNonce(caller) : 0);
```

Comment:
it seems like half of the logic got extracted and now it is split between two classes